ultimatepython/classes/encapsulation.py

```diff
@@ -0,0 +1,117 @@
+"""    
+    Encapsulation is a feature of OOP that allows you to hide the
+    implementation details of a class from its users.
+    Encapsulation allows us to limit the access of certain attributes 
+    within a class. This prevents users from directly accessing and modifying such
+    attributes from outside the class. Instead, users must use methods to access and
+    modify attributes.
+"""
+import random
+
+class BankAccount:
+    def __init__(self, account_holder_name: str):
+        """
+            In python, a class attribute can be made private by prefixing it with two underscores.
+            This makes it inaccessible to users outside the class.
+            By default, class variables are public. Therefore, they can be accessed and modified
+            outside of the class.
+            
+            Here, account_number and balace are private while account_holder_name is public.
+         """
+        self.account_holder_name: str = account_holder_name
+        """
+            The account number is generated automaticallyusing the randint function from the random module
+            when a new instance of the class is created.
+            The balance is set to 0.0 by default.
+        """
+        self.__account_number: int = random.randint(1000000000, 9999999999) # generate a random account number of 10 digits.
```

Comment:
This should raise a `flake8` issue because the comment isn't two-spaces apart from the actual code